hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java

```diff
@@ -446,4 +448,22 @@ private static int findKeyIndex(Table table) {
         }
         return null;
     }
+
+    /**
+     * Return true if the `expression` contains any input reference to a field
+     * with index in `indexes`.
+     */
+    public static boolean hasInputRef(RexNode expression, int... indexes) {
+        boolean[] res = {false};
+        expression.accept(new RexVisitorImpl<Void>(true) {
+            @Override
+            public Void visitInputRef(RexInputRef inputRef) {
+                if (arrayIndexOf(inputRef.getIndex(), indexes) >= 0) {
+                    res[0] = true;
```

Comment:
Any reason this has to be an array? Just checking.

Reply:
@ivanthescientist outer variable should be effective final to be accessible within the anonymous class/lambda context